ops/model.py

```diff
@@ -1676,6 +1694,20 @@ def get_pebble(self, socket_path: str) -> 'pebble.Client':
         """Create a pebble.Client instance from given socket path."""
         return pebble.Client(socket_path=socket_path)
 
+    def planned_units(self) -> int:
+        """Count of "planned" units that will run this application.
+
+        We use goal-state here, in the simplest possible way. When we implement goal state
+        more completely, this should turn into a wrapper around the more complete call to
+        goal state.
+
+        Includes the current unit in the count.
+
+        """
+        goal_state = self._run('goal-state', return_output=True, use_json=True)
```

Comment:
Might also be worth adding a comment before that line so future readers can understand that perspective. Something along the lines of:
```
# goal-state as a concept in juju is dying in favor of it being simply the current state,
# so we must not use this API further outside of explicitly designed and agreed cases.
```